clients/src/main/java/org/apache/kafka/clients/CommonClientConfigs.java

```diff
@@ -203,4 +205,15 @@ public static Map<String, Object> postProcessReconnectBackoffConfigs(AbstractCon
         }
         return rval;
     }
+
+    public static void postValidateSaslMechanismConfig(AbstractConfig config) {
+        SecurityProtocol securityProtocol = SecurityProtocol.forName(config.getString(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG));
+        String clientSaslMechanism = config.getString(SaslConfigs.SASL_MECHANISM);
+        if (securityProtocol == SecurityProtocol.SASL_PLAINTEXT || securityProtocol == SecurityProtocol.SASL_SSL) {
+            if (clientSaslMechanism == null || clientSaslMechanism.trim().isEmpty()) {
```

Comment:
Nit: We don't need to `trim` here, do we?
```suggestion
            if (clientSaslMechanism == null || clientSaslMechanism.isEmpty()) {
```

Reply:
Strings containing only spaces are also considered illegal values, as indicated in the corresponding testCase.

```
        configs.put(SaslConfigs.SASL_MECHANISM, " ");
        ce = assertThrows(ConfigException.class, () -> new ConsumerConfig(configs));
```

Reply:
Isn't the string automatically trimmed here by the `AbstractConfig`?

Reply:
Oh, you are right. The `ConfigDef.parseType` method already does this logic.
Thanks.